src/app.py

```diff
@@ -37,6 +37,19 @@ def index():
         "index.html", user=current_user, items=items, category_groups=category_groups
     )
 
+@app.route("/product_info/<int:serial_number>", methods=["POST", "GET"])
+@login_required
+def product_info(serial_number):
+    if session.get("logged_in", False) == False:

```

Comment:
This is not needed since we know that the user is logged in from the @login_required tag

Reply:
Fixed.

Reply:
We should keep the login required, remove the if statement

Reply:
Fixed.